util/util_test.go

```diff
@@ -0,0 +1,47 @@
+package util
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCallWithTimeout_timeout(t *testing.T) {
+	done := make(chan struct{})
+	defer func() {
+		close(done)
+	}()
+
+	logger, hook := test.NewNullLogger()
+	err := CallWithTimeout(context.Background(), logger, 1*time.Nanosecond, func(ctx context.Context) error {
+		<-done
+		return errors.New("should not return")
+	})
+
+	require.Error(t, err, "There should be an error")
+	require.ErrorIs(t, err, ErrTimeout)
+
+	time.Sleep(2 * time.Second)
```

Comment:
This is also testing the "misbehaving handler" log statement. Since that is hardcoded to 1 second after the timeout this sleep must be in the 1-2 second range. I put it at 2 seconds to ensure it always passes.